src/encoder/mod.rs

```diff
@@ -584,27 +580,27 @@ impl<'a, W: 'a + Write + Seek, T: ColorType> ImageEncoder<'a, W, T> {
         width: u32,
         height: u32,
     ) -> TiffResult<ImageEncoder<'a, W, T>> {
-        let row_samples = u64::from(width) * <T>::BITS_PER_SAMPLE.len() as u64;
+        let row_samples = u64::from(width) * u64::try_from(<T>::BITS_PER_SAMPLE.len())?;
```

Comment:
Note for review readers: Reasoning would not be the same here, since `T` is an input type parameter. Panicking or build failures are not exactly welcome so keep the fallible conversion, imo.